ballerina/utils.bal

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) 2022 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/jballerina.java;
+import ballerina/lang.'string as strings;
+
+isolated function getSubstring(string str, int startIndex) returns string {
+    if startIndex == 0 {
+        return str;
+    }
+    return strings:substring(str, startIndex, str.length());
+}
+
+isolated function getReplacementString(string str, string regex, Replacement replacement, int startIndex = 0)
+                            returns string|error {
+    if replacement is string {
+        return replacement;
+    } else {
+        Match?|error matched = trap search(str, regex, startIndex);
+        if matched is Match {
+            return replacement(matched);
+        } else {
+            return error("There is no matching substrig in the given string.");
+        }
+    }
+}
```

Comment:
Is this replacement string logic correct?

Reply:
This logic is calling [here](https://github.com/ballerina-platform/module-ballerina-regex/pull/377/files#diff-f91d32ea125a036a099dbf0796354f47e62985529bbdc4f75638414169f49910R47). If this returns an error, The `replace` function returns the original string.

Reply:
I added a comment in the usage. Can't we use the same logic as `replaceAll`? We can avoid using java function

Reply:
Updated now.